### ISSUE TITLE
List Tenants REST API is not filtering by authUserId

### ISSUE THREAD

**Issue by jorgevillaverde-sitewhere** [sitewhere/sitewhere#902]
When calling List Tenant and adding `authUserId` equals `prueba`

```console
curl --location --request GET 'http://<host>:<port>/sitewhere/api/tenants?authUserId=prueba&includeRuntimeInfo=false&page=1&pageSize=100' \
--header 'Authorization: <token>'
```

The service is not filtering by `authUserId` and returning all tenants. 
This is the result: 

```json
{
    "numResults": 2,
    "results": [
        {
            "metadata": null,
            "token": "default",
            "createdDate": "2020-11-12T13:40:18.000+0000",
            "borderColor": null,
            "imageUrl": "https://s3.amazonaws.com/sitewhere-demo/sitewhere-small.png",
            "icon": null,
            "authenticationToken": "sitewhere1234567890",
            "authorizedUserIds": [
                "admin",
                "noadmin"
            ],
            "configurationTemplateId": "default",
            "datasetTemplateId": "construction",
            "backgroundColor": null,
            "foregroundColor": null,
            "name": "Default Tenant"
        },
        {
            "metadata": null,
            "token": "tenant",
            "createdDate": "2020-11-12T14:46:08.000+0000",
            "borderColor": null,
            "imageUrl": null,
            "icon": null,
            "authenticationToken": "tenant",
            "authorizedUserIds": [
                "prueba"
            ],
            "configurationTemplateId": "default",
            "datasetTemplateId": "empty",
            "backgroundColor": null,
            "foregroundColor": null,
            "name": "tenant"
        }
    ]
}
```

**Comment by kaleemulhaque** [comment]
i am also facing the same problem.if you got the solutions or not

**Comment by derekadams** [comment]
This is fixed in `v3.0.5`. It also guards get/update/delete on tenants where the logged in user is not in the list.
